src/Control/Effect/Empty.hs

```diff
@@ -32,7 +38,8 @@ guard True  = pure ()
 guard False = empty
 
 
--- $setup
--- >>> :seti -XFlexibleContexts
--- >>> import Test.QuickCheck
--- >>> import Control.Carrier.Empty.Maybe
+-- Properties
+
+-- | 'empty' annihilates '>>='.
+empty_annihilation :: Has Empty sig m => (m b -> m b -> prop) -> (a -> m b) -> prop
+empty_annihilation (===) k = (empty >>= k) === empty
```

Comment:
We additionally define and export properties parameterized by the equality test. This property can be used with QuickCheck, hedgehog, and presumably other property testing frameworks, and doesn’t require the library to depend on any of them.

The intention is for every `Empty` carrier to test its conformance to this law.